src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceGenerator.cs

```diff
@@ -48,72 +48,71 @@ public sealed class SourceGenerator : CachingSourceGenerator
             defaultSeverity: DiagnosticSeverity.Error,
             isEnabledByDefault: true);
 
-        protected override bool TryGetRelevantInput(in GeneratorExecutionContext context, out AdditionalText? input, out SourceText? inputText)
+        public void Initialize(IncrementalGeneratorInitializationContext context)
         {
-            input = context.AdditionalFiles.SingleOrDefault(a => Path.GetFileName(a.Path) == "Syntax.xml");
-            if (input == null)
-            {
-                context.ReportDiagnostic(Diagnostic.Create(s_MissingSyntaxXml, location: null));
-                inputText = null;
-                return false;
-            }
+            var syntaxXmlFiles = context.AdditionalTextsProvider.Where(at => Path.GetFileName(at.Path) == "Syntax.xml").Collect();
 
-            inputText = input.GetText();
-            if (inputText == null)
+            context.RegisterSourceOutput(syntaxXmlFiles, static (context, syntaxXmlFiles) =>
             {
-                context.ReportDiagnostic(Diagnostic.Create(s_UnableToReadSyntaxXml, location: null));
-                return false;
-            }
+                var input = syntaxXmlFiles.SingleOrDefault();
```

Comment:
checking my understanding: this factoring (putting the 'Where' before 'Collect', instead of as an argument here) is what makes it so we only re-run on changes to Syntax.xml specifically, right?

Reply:
That was one bit I wasn't entirely sure about actually: @chsienki which way works better for caching here?

Reply:
Both ways work. Incremental generators don't provide "all or nothing" execution; they execute the steps up until the point that the output of a step produces the same output as the previous execution. The main advantage of `Where` before `Collect` is it avoids the need to provide a collection comparer to handle the comparisons for the output of `Where`.

Reply:
Yeah that was my assumption for doing it this way, but I wasn't sure if I needed the comparer either way.

Reply:
> they execute the steps up until the point that the output of a step produces the same output as the previous execution

Thanks, this description made things much clearer to me.